### PR TITLE
Add install target to CMakeLists.txt

### PR DESCRIPTION
**Pull request by JonnyPtn**
To allow installation when using cmake, e.g:

```
cmake .
make install
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,4 +71,12 @@ add_library(enet STATIC
...
-endif()
\ No newline at end of file
+endif()
+
+install(TARGETS enet
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION lib/static
+    LIBRARY DESTINATION lib)
+
+install(DIRECTORY include/
```
Comment (kdavison) [inline CMakeLists.txt:81]: +1 to this commit in general!

after some research on my own I believe the "accepted" way to include headers is:

set_target_properties(enet PROPERTIES PUBLIC_HEADER "${INCLUDE_FILES}")
install(
	TARGETS enet
	LIBRARY DESTINATION lib
	PUBLIC_HEADER DESTINATION include/enet
)

Its cmake so maybe there is some side effect to PUBLIC_HEADERS or some advantage to your method that I am not aware of.

**Comment by JayFoxRox** [comment]
- Please rebase? @JonnyPtn 
- Please review? @lsalzman

Having a CMake install target would help with abstracting the build-system complexity away (makes it easy to use enet for projects with / without CMake and system-wide.. as long as CMake exists for the target platform, which is easy to port, thanks to toolchain files).
I could imagine that many other platforms would also benefit from this.

**Comment by JonnyPtn** [comment]
Rebased :+1:

**Comment by JonnyPtn** [comment]
@lsalzman Anything more I can do with this? trying to clear up my pending PR lists so if you don't want it feel free to just close it

**Pull request merged**